### DISCUSSION TITLE
get_message_from_id/get_message_info_from_id

### DISCUSSION THREAD

**Discussion by explogeek** [python-telegram-bot/python-telegram-bot#4143] [Feature Suggestions]
Currently, there is no stable and easy way to get a copy of a message from message_id. Forwarding is a crutch, also doesn't work in closed groups with no copying allowed. I need to get info about message, but can't do it from just the message_id.

```python
# this works only for messages with no reply, because for some reason, 
# if the message is a reply, chat.pinned_message will return the reply instead of the pinned message
async def get_message_from_id(id, chat_id, bot):
    await bot.pin_chat_message(chat_id, id, True)
    chat = await bot.get_chat(chat_id)
    message = chat.pinned_message
    await bot.unpin_chat_message(chat_id, id)
    return message
```
```python
# this only works for places where you are allowed to forward messages. 
# copying does not return Message object, so it's useless in this case
async def get_message_text_from_id(id, chat_id, bot) -> str:
    message = await bot.forward_message(chat_id, chat_id, id, disable_notification=True)
    text = message.gtext
    await message.delete()
    return text
```
```python
# gtext is a crutch I use, monkey patching Message class
def get_gtext(self):
    return self.text if self.text else self.caption

Message.gtext = property(get_gtext)
```

**Comment by Poolitzer** [comment]
This can not be added by PTB and would need to be added by the bot API. See https://github.com/tdlib/telegram-bot-api/issues/62#issuecomment-747456729 on why they probably will never add it.

You already have the two crude workarounds (you can also try to reply to a message id and get its content that way).

**Comment by Bibo-Joshi** [comment]
Hi. This functionality is simply not provided by the Bot APi. See also https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-Asked-Questions#can-you-add-feature-to-ptb-can-i-do-thing-with-my-bot.
